backend/resources/voucher.py

```diff
@@ -94,7 +95,26 @@ def voucher_post_institution(session, user_inst):  # pylint:disable=unused-argum
                            institution_id=institution_id,
                            )
 
-    # ToDo Blockchain
+    institution_sc = WEB3.eth.contract(
+        address=res.scAddress,
+        abi=INSTITUTION_JSON["abi"]
+    )
+
+    if int(voucher_valid_time) - time.time() < 0:
+        return jsonify({'error': 'validTime is smaller than current time'}), 400
+
+    voucher_description_bytes = str.encode(voucher_description)
+    transaction = institution_sc.functions.addVoucher(user_inst.publickeyUser, voucher_description_bytes,
+                                                      int((int(voucher_valid_time) - time.time()) / 86400))
+    transaction = transaction.buildTransaction({'nonce': WEB3.eth.getTransactionCount(user_inst.publickeyUser),
+                                                'from': user_inst.publickeyUser})
+    signed_transaction = WEB3.eth.account.sign_transaction(transaction, user_inst.privatekeyUser)
+
+    tx_hash = WEB3.eth.sendRawTransaction(signed_transaction.rawTransaction)
+    tx_receipt = WEB3.eth.waitForTransactionReceipt(tx_hash)
+
+    if tx_receipt.status != 1:
+        raise RuntimeError("SC Call failed!")
```

Comment:
macht das hier überhaupt sinn?
Erstellst du hier nicht für den Institutionsbesitzer einen Gutschein?

Reply:
Ich hoffe doch das es sinn macht.. ich dachte hier wird einfach die addVoucher funktion aufgerufen auf dem zur Institution gehörenden smart contract und das kann nur der Besitzer bzw. ein admin machen

Reply:
achso stimmt, es muss die addresse vom Besitzer des Gutscheins mit übergeben werden und anstelle von user_inst.publickeyUser in Zeile 107 weiter gegeben werden, richtig?
Wie wird die addresse denn am besten übergeben? Auch als header?

Reply:
Der SC muss hier garnicht aufgerufen werden, im grunde genommen muss hier nur das ToDo entfernt werden 😔

Reply:
Ach als ob.. ist das issue dann fertig? Also war das ein fehler oder bezieht sich das auf was anderes?

Reply:
war nen fehler, sry